### PR TITLE
Refactor and move binder code to BinderUtils

### PR DESCRIPTION
**Pull request by bryancroteau-MSFT**
Clean up, rename, and move Binary and Unary node binder code into BinderUtils.

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs [hunk 1]
```diff
@@ -3943,47 +3665,13 @@ private bool UsesParentsNestedAwareProperty(IExternalControl controlInfo, DName
...
-            }
-
             public override void PostVisit(UnaryOpNode node)
             {
                 AssertValid();
 
                 var childType = _txb.GetType(node.Child);
 
-                var res = PostVisitCore(_txb.ErrorContainer, node, childType);
+                var res = CheckUnaryOpCore(_txb.ErrorContainer, node, childType);
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:3674]: >CheckUnaryOpCore [](http://example.com/codeflow?start=26&length=16)

BinderUtils.CheckUnaryOpCore?

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs (hunk 1 above)
Comment (bryancroteau-MSFT) [reply src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:3674]: All the BinderUtils symbols are being statically imported.

**Pull request merged**